File: corelibrary/src/main/java/pers/cxd/corelibrary/base/FragmentCreator.java
package pers.cxd.corelibrary.base;

import androidx.fragment.app.Fragment;

/**
 * A Creator to create fragment by {@code fmtClazz}
 *
 * @param <T> Fragment type
 * @author pslilysm
 * @since 1.0.5
 */
public interface FragmentCreator<T extends Fragment> {

    T create(Class<T> fmtClazz);

}

File: corelibrary/src/main/java/pers/cxd/corelibrary/base/FragmentCreator.kt
package pers.cxd.corelibrary.base

import androidx.fragment.app.Fragment

/**
 * A Creator to create fragment by `fmtClazz`
 *
 * @param <T> Fragment type
 * @author pslilysm
 * @since 1.0.5
</T> */
interface FragmentCreator<T : Fragment?> {
    fun create(fmtClazz: Class<T>): T
}